syntax = "proto3";

service AccountService {
    rpc CreateAccount (AccountRequest) returns (AccountResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc GetProfile (ProfileRequest) returns (ProfileResponse);
}

message AccountRequest {
    string username = 1;
    string password = 2;
    string email = 3;
}

message AccountResponse {
    string status = 1;
    string message = 2;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string status = 1;
    string token = 2;
}

message ProfileRequest {
    string token = 1;
}

message ProfileResponse {
    string username = 1;
    string email = 2;
    string status = 3;
}
